### ISSUE TITLE
[Bug] RuntimeError - Early iteration NaN gradient for Profet HPO benchmarks

### ISSUE THREAD

**Issue by hvarfner** [pytorch/botorch#945]
# 🐛 Bug

Hi,

I've been trying to run experiments with the Profet (https://arxiv.org/abs/1905.12982) HPO surrogate benchmarks on what I hope is a vanilla BOTorch setup with double precision on CPU. When I run the synthetic benchmarks provided by BOTorch (Hartmann/Branin etc.), the performance is fine and no NaN issues. However, when I run the Profet benchmarks (SVM specifically, but also XGBoost), NaN gradients pop up in a seemingly stochastic fashion. These surrogate benchmarks are noiseless.

## To reproduce
https://github.com/hvarfner/profetissue
May need to install pybnn for Profet (sorry), otherwise requirements are in repo. 


## To run:
python main -f hartmann (should run fine and just as normal.)
python main -f svm (Profet benchmark, should produce RuntimeError after 10-50 iterations about half the time).

## Output:

**Iterations (no duplicates in design points):**
Initialization: Sampled from SOBOL:  :
                                                                                               [19/1363]
tensor([[0.9975, 0.1044],                                                                                                                    
        [0.1111, 0.7880],                                                                                                                    
        [0.3652, 0.3533]], dtype=torch.float64)                                                                                                                                                                                                             
4) Next point: [[0.6314828251231542, 0.054613702723228984]]  ---  Best value: -2.77e-02                                                      
5) Next point: [[0.8089566363614312, 0.30184720174101265]]  ---  Best value: -2.42e-02                                                       
6) Next point: [[0.0257129078058237, 0.0]]  ---  Best value: -2.42e-02                                                                       
7) Next point: [[0.781244838715517, 0.14944733516389283]]  ---  Best value: -2.42e-02                                                        
8) Next point: [[0.5952470084548915, 0.31611042705286674]]  ---  Best value: -2.42e-02                                                       
9) Next point: [[1.0, 0.623351174330139]]  ---  Best value: -2.42e-02                                                                        
10) Next point: [[0.8773703274595368, 0.0]]  ---  Best value: -2.42e-02                                                                      
11) Next point: [[0.7128756825235761, 0.2473001320729077]]  ---  Best value: -2.42e-02                                                       
12) Next point: [[0.8209517191002765, 0.059517805662041845]]  ---  Best value: -2.36e-02                                                     
13) Next point: [[0.9036311911359326, 0.20907373799339207]]  ---  Best value: -2.36e-02                                                      
14) Next point: [[1.0, 0.0]]  ---  Best value: -2.36e-02                                                                                     
15) Next point: [[0.7033234932648541, 0.3641839555036253]]  ---  Best value: -2.36e-02

## Stack trace/error message 
Traceback (most recent call last):                                                                                                           
  File "/home/hvarfner/Documents/PriorBO/models/pibo_botorch/main.py", line 97, in <module>                                                  
 new_x_ei, new_obj_ei = optimize_acqf_and_get_observation(acq_function, fun)
  File "/home/hvarfner/Documents/PriorBO/models/pibo_botorch/main.py", line 49, in optimize_acqf_and_get_observation
    candidates, _ = optimize_acqf(
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/botorch/optim/optimize.py", line 184, in optimize_acqf
    batch_candidates_curr, batch_acq_values_curr = gen_candidates_scipy(
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/botorch/generation/gen.py", line 166, in gen_candidates_scipy
    res = minimize(
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/_minimize.py", line 623, in minimize
    return _minimize_lbfgsb(fun, x0, args, jac, bounds,
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/lbfgsb.py", line 360, in _minimize_lbfgsb
    f, g = func_and_grad(x)
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py", line 267, in fun_and_grad
    self._update_fun()
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py", line 233, in _update_fun
    self._update_fun_impl()
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py", line 137, in update_fun
    self.f = fun_wrapped(self.x)
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py", line 134, in fun_wrapped
    return fun(np.copy(x), *args)
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/optimize.py", line 74, in __call__
    self._compute_if_needed(x, *args)
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/scipy/optimize/optimize.py", line 68, in _compute_if_needed
    fg = self.fun(x, *args)
  File "/home/hvarfner/anaconda3/envs/botorch/lib/python3.9/site-packages/botorch/generation/gen.py", line 162, in f
    raise RuntimeError(msg)
RuntimeError: 10 elements of the 10 element gradient array `gradf` are NaN. This often indicates numerical issues.

## Expected result:
The global min of **0.0218** is reached in a good number of cases when the run actually finishes. Something close to that should be the case almost every single time. One of the other benchmarks (xgboost) behaves similarly, but not as dramatic. This should reach a final value of about 9 to 9.5.


**System information**

**Versions:**
- BOTorch 0.5.1
- GPyTorch 1.5.1
- PyTorch 1.9.0
- Ubuntu 18.04

## Additional context

**Comment by saitcakmak** [comment]
Hi @hvarfner. I looked through the code and noticed that you're using `FixedNoiseGP` with `train_Yvar = 0`. This is in general a bad idea and will lead to numerical issues. Adding a small bit of `train_Yvar`, even when the observations are noise free, will help get around these issues. I'd recommend using a small noise level like `1e-6 / 1e-4`. I ran your code without any issues after replacing `NOISE_SE = 0` with `NOISE_SE = 1e-6`.
